packages/babel-plugin-transform-es2015-block-scoping/test/fixtures/general/switch/expected.js

```diff
@@ -9,3 +11,17 @@ switch (a) {
     var d = 3;
     break;
 }
+
+switch (true) {
+  case true:
+    var _a = false;
+    var _b = true;
+    var _c = 4;
+    var _d = 5;
+
+  case false:
+    class _e {}
```

Comment:
Class declarations are block-scoped just like `let`/`const`.

     class e {}

is essentially 

    let e = class {}

but looks like it doesn't take that into account.